src/python_bindings/py_algorithm.h

```diff
@@ -3,13 +3,53 @@
 #include <memory>
 #include <type_traits>
 #include <unordered_map>
+#include <vector>
 
 #include <pybind11/pybind11.h>
 #include <pybind11/stl.h>
 
 #include "algorithms/algorithm.h"
+#include "algorithms/metric/enums.h"
+#include "config/equal_nulls/type.h"
+#include "config/error/type.h"
+#include "config/indices/type.h"
+#include "config/max_lhs/type.h"
 #include "config/tabular_data/input_table_type.h"
+#include "config/thread_number/type.h"
+#include "model/types/types.h"
+namespace {
+using ConvFunction = std::function<std::string(boost::any)>;
+template <typename T>
+std::pair<std::string, ConvFunction> generate_conv_pair(std::string name) {
+    return std::pair<std::string, ConvFunction>{
+            name, [](boost::any value) { return std::to_string(boost::any_cast<T>(value)); }};
+}
+
+template <>
+std::pair<std::string, ConvFunction> generate_conv_pair<bool>(std::string name) {
+    return std::pair<std::string, ConvFunction>{
+            name, [](boost::any value) { return boost::any_cast<bool>(value) ? "True" : "False"; }};
+}
+const std::unordered_map<std::string, ConvFunction> converters{
+        generate_conv_pair<config::EqNullsType>("is_null_equal_null"),
+        generate_conv_pair<bool>("has_tid"),
+        generate_conv_pair<bool>("dist_from_null_is_infinity"),
+        generate_conv_pair<config::ThreadNumType>("threads"),
+        generate_conv_pair<config::MaxLhsType>("max_lhs"),
+        generate_conv_pair<config::ErrorType>("error"),
+        generate_conv_pair<int>("seed"),
+        generate_conv_pair<unsigned int>("item_column_index"),
+        generate_conv_pair<double>("minconf"),
+        generate_conv_pair<double>("minsup"),
+        generate_conv_pair<unsigned int>("tid_column_index"),
+        generate_conv_pair<long double>("parameter"),
+        generate_conv_pair<unsigned int>("q"),
+};
 
+std::string opt_to_str(std::string_view name, boost::any val) {
```

Comment:
What I meant by "similar to `PyToAny`" is getting the any object, then converting it to `py::object`, based on the `type_index` the algorithm returns.
I'm looking for a `std::type_index -> std::function<py::object(boost::any)>` mapping.